src/dict_mem.cpp

```diff
@@ -744,9 +749,18 @@ int DictMem::GetRootEdge(int nt, EdgePtrs &edge_ptrs) const
 // The temp edge is written to shared memory for handling segfault situations.
 // When writer restarts from segfault, it will retry WriteEdge so that the DB is
 // maintained consistently.
-int DictMem::GetRootEdge_Writer(int nt, EdgePtrs &edge_ptrs) const
+int DictMem::GetRootEdge_Writer(bool rc_mode, int nt, EdgePtrs &edge_ptrs) const
 {
-    edge_ptrs.offset = root_offset + NODE_EDGE_KEY_FIRST + NUM_ALPHABET + nt*EDGE_SIZE;
+    if(rc_mode)
+    {
+        if(root_offset_rc == 0)
+            abort();
```

Comment:
Should throw exception instead of abort()

Reply:
done